coverage_comment/github.py

```diff
@@ -54,7 +54,14 @@ def download_artifact(
     filename: pathlib.Path,
 ) -> str:
     repo_path = github.repos(repository)
-    artifacts = repo_path.actions.runs(run_id).artifacts.get().artifacts
+    page = 1
+    artifacts = []
+    while True:
+        result = repo_path.actions.runs(run_id).artifacts.get(page=str(page))
+        if not result:
+            break
+        artifacts.extend(result.artifacts)
+        page += 1
     try:
         artifact = next(
```

Comment:
What about incorporating reading the pages and paginating into one generator (`yield`), so that if you find the artifact on the first page, you don't need to fetch the second page ? Also, this will limit the number of tests you have to change while still letting you do dedicated tests for cancellation.

Also, if the number of elements github returns on a page is fewer than the max, you know in advance there won't be a next page,

Reply:
you're right, I'll work on improving it

Reply:
To avoid unnecessarily fetching the next page, I leveraged on the `total_count` attribute of the response, instead of using the github default max page size of 30, it seems safer to me